.github/workflows/ci.yaml

```diff
@@ -0,0 +1,281 @@
+name: CI Tasks for Ory Hydra
+on:
+  push:
+    branches:
+      - main
+      - master
+    tags:
+      - '*'
+  pull_request:
+
+jobs:
+  test:
+    name: Run tests and lints
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' || github.ref_name == 'master' }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ^1.13
+      - run: go list -json > go.list
+      - name: Run nancy
+        uses: sonatype-nexus-community/nancy-github-action@v1.0.2
+      - name: Run prettier
+        uses: ory/ci/prettier@master
+      - name: Run prettier (docs)
+        uses: ory/ci/prettier@master
+        with:
+          dir: docs
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        env:
+          GOGC: 100
+        with:
+          args: --timeout 10m0s --issues-exit-code=0
+          skip-go-installation: true
+          skip-pkg-cache: true
+      - run: |
+          make .bin/go-acc
+          .bin/go-acc -o coverage.out ./... -- -failfast -timeout=20m -tags sqlite
+      - run: |
+          bash <(curl -s https://codecov.io/bash)
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+
+  test-hsm:
+    name: Run HSM tests
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' }}
+    env:
+      HSM_ENABLED: true
+      HSM_LIBRARY: /usr/lib/softhsm/libsofthsm2.so
+      HSM_TOKEN_LABEL: hydra
+      HSM_PIN: 1234
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ^1.17
+      - run: |
+          sudo apt install -y softhsm opensc
+          sudo rm -rf /var/lib/softhsm/tokens
+          sudo mkdir -p /var/lib/softhsm/tokens
+          sudo chmod -R a+rwx /var/lib/softhsm
+          sudo chmod a+rx /etc/softhsm
+          sudo chmod a+r /etc/softhsm/*
+          pkcs11-tool --module /usr/lib/softhsm/libsofthsm2.so --slot 0 --init-token --so-pin 0000 --init-pin --pin 1234 --label hydra
+          go test -p 1 -v -failfast -short -timeout=20m -tags=sqlite,hsm ./...
+
+  test-e2e:
```

Comment:
Can we split this test into multiple subtest so that all of these run in parallel? One problem here will be that `./test/e2e/circle-ci.bash` does a lot of bootstrapping (npm i, go build, ...). Maybe it is possible to prep the run once and then run the `run` steps in parallel?

Reply:
Understandable. Let me see what I can do.

Reply:
PR